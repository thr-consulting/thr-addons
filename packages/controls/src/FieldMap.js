// @flow

import React from 'react';

type Props = {
	value: any,
	as: Function,
	asValue: string,
	onChange: Function,
};

/**
 * Maps onChange handler parameters from (event, data) to just (data).
 * @class
 * @property {*} value - The value of the field
 * @property {Component} as - The React Component to render
 * @property {string} asValue
 * @property {onChange} onChange - The normal onChange handler
 */
export default function FieldMap({value, as, asValue, onChange, ...rest}: Props) {
	const props = {
		...rest,
		onChange: (e, data) => onChange(data[asValue]),
	};
	props[asValue] = value;

	return React.createElement(as, props);
}
