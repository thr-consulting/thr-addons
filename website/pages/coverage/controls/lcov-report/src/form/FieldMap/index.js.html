<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/form/FieldMap/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="/lcov/prettify.css" />
    <link rel="stylesheet" href="/lcov/base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(/lcov/sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/form/FieldMap</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export {default} from './FieldMap';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 03 2018 16:58:16 GMT-0500 (CDT)
</div>
</div>
<script src="/lcov/prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="/lcov/sorter.js"></script>
<script src="/lcov/block-navigation.js"></script>
</body>
</html>
