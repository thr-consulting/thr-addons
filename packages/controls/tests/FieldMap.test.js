import React from 'react';
import {shallow, mount} from 'enzyme';
import {Input} from 'semantic-ui-react';
import FieldMap from '../src/FieldMap';

describe('FieldMap', () => {
	it('should render without throwing an error', () => {
		const mockOnChange = jest.fn();

		expect(shallow(
			<FieldMap
				value="a"
				as={Input}
				asValue="value"
				onChange={mockOnChange}
			/>)).toMatchSnapshot();
	});

	it('should respond to changes', () => {
		const mockOnChange = jest.fn();

		const wrapper = mount(
			<FieldMap
				value="initial value"
				as={Input}
				asValue="value"
				onChange={mockOnChange}
			/>);
		const u = wrapper.find('input');
		u.simulate('change', {target: {value: 'blah blah'}});

		expect(mockOnChange.mock.calls.length).toBe(1);
		expect(mockOnChange.mock.calls[0][0]).toEqual('blah blah');
	});
});
