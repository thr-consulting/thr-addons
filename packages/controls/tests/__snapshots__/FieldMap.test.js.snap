// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldMap should render without throwing an error 1`] = `
<Input
  onChange={[Function]}
  type="text"
  value="a"
/>
`;
